src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,7 +579,7 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
 
         private void OnEnvDTEProjectAdded(EnvDTE.Project envDTEProject)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
```

Comment:
I couldn't repro it, can you give me exact scenario or if you have some test code that will be great. Since creating a new emoty solution doesn't ever take much time so add project is always able to add project in nuget cache. If you can give me the repro then I'll try to take care of that as well.